DeepSpeech.py

```diff
@@ -309,8 +309,21 @@ def get_tower_results(iterator, optimizer, dropout_rates):
                     tower_avg_losses.append(avg_loss)
 
                     # Compute gradients for model parameters using tower's mini-batch
-                    gradients = optimizer.compute_gradients(avg_loss)
-
+                    if FLAGS.load == "transfer" and not FLAGS.fine_tune:
```

Comment:
It'd be better to move this check to the `train` function and then just pass `drop_source_layers=None` for the case where we should not do any layer dropping. That way all of the logic around the behavior of the CLI flags is contained to `train`.